### ISSUE TITLE
codenode with django 1.3 (status?)

### ISSUE THREAD

**Issue by mhoffman** [codenode/codenode#64]
Hello,

thank bunches for codenode. It seems like a really nice way to demonstrate APIs/Python/Sage to beginners. Though installing the current version in combination with django 1.3 makes me wonder what the current development status is. Besides getting a bunch of deprecation warning, every attempt evaluations end with

/lib/python2.6/site-packages/codenode-0.03-py2.6.egg/codenode/frontend/async/backend.py", line 219, in _success
        data['cellid'] = cellid
    exceptions.TypeError: 'NoneType' object does not support item assignment

I would not mind getting involved in the project so any pointers to how much work would be involved to get it up and running with recent versions of django would be much appreciated.

Best,
Max.

**Comment by jamescasbon** [comment]
Hi Max, 

Codenode is not that actively maintained at the moment due to my dereliction.  Are you using the current version?  If so you may be the only person!

ipython notebook may be a maintained alternative that might do what you want.  Otherwise, I will try and have a look at 1.3 supoprt.  Is this something you would be wiling to help with?

**Comment by mhoffman** [comment]
Oh, fast feedback! Give me a day or so to look into it. I think it would be quit good to have a pluggable shell for a number science related projects. Can't promise a lot though. 

Greetings from Chicago,
Max.

**Comment by jamescasbon** [comment]
Yes, I agree codenode was the right idea here, but to be honest the combination of twisted, django and javascript frontend was quite a lot to maintain for me.

I have started a rewrite of the frontend to use some javascript frameworks to help me out (ie backbone) which aims to be a pluggable shell for multiple code engines.

If I get 30 mins, I will try and run codenode on 1.3 and see if I can work out what is going on.
